ml-agents/mlagents/trainers/trainer_controller.py

```diff
@@ -234,12 +235,15 @@ def start_learning(self, env_manager: EnvManager) -> None:
             KeyboardInterrupt,
             UnityCommunicationException,
             UnityEnvironmentException,
+            UnityCommunicatorStoppedException,
         ) as ex:
             self.kill_trainers = True
             if self.train_model:
                 self._save_model_when_interrupted()
 
-            if isinstance(ex, KeyboardInterrupt):
+            if isinstance(ex, KeyboardInterrupt) or isinstance(
+                ex, UnityCommunicatorStoppedException
+            ):
                 pass
             else:
                 # If the environment failed, we want to make sure to raise
```

Comment:
We should still make sure the `.nn` is always saved, right?

Reply:
Added `finally` block

Reply:
Hang on, I'm not convinced that the `finally` will happen, since you're raising in the except.

Might be easier to do
```
if self.train_model:
    self._export_graph()
if isinstance(...):
    pass
else:
    raise ex
```

Reply:
Never mind, I'm convinced. Carry on.